nbdev/config.py

```diff
@@ -67,7 +67,8 @@ def _apply_defaults(
     if copyright is None: copyright = f"{datetime.now().year} ownwards, %(author)s"
     for k,v in locals().items():
         if k.startswith('_') or k == 'cfg' or cfg.get(k) is not None: continue
-        cfg[k] = v
+        if k == 'lib_name': cfg[k] = cfg.repo.replace('-', '_') # the repo name might have dashes but the lib name cannot
```

Comment:
Shouldn’t lib_name default to ‘%(repo)s’ as it used to? Dashes are fine in a package name just not in the importable package itself, which is lib_path (already handled above)